### PR TITLE
Add support for Axum framework

### PR DESCRIPTION
**Pull request by brunoribeiro127**
This MR adds support for the Axum framework.

### CODE REVIEW

**Review by DarrenBaldwin07 — COMMENTED** [review]
Thank you so much for the PR @brunoribeiro127 - taking a first pass today and will come back with some review comments.

**Comment by capveg-netdebug** [comment]
Nice PR @brunoribeiro127  - what's the status of this?  I'd love to use this code as I'm also working with Axum.  I'm still reading through the changes but it doesn't look like you used https://docs.rs/axum-login/latest/axum_login/ - any reason?

**Comment by brunoribeiro127** [comment]
@capveg-netdebug Thanks for reviewing. To be honest, I was not aware of axum_login at all. It seems like the best way forward!

**Comment by capveg-netdebug** [comment]
Thanks for the quick reply.  If that's the case, I might try to push some of my axum_login integration here for folks to look at/think about.  It might be that my use-case is a bit unique in that I want to include some non-trivial state with the user identity that's stored locally to my app and not in Clerk but I'm curious how you're thinking about it as it affects the design.

**Review by DarrenBaldwin07 — COMMENTED** [review]
I'd like to get this in too! I personally havent used axum much so not sure what best practices are here. I do like the idea of having a central authorizer that then gets consumed by the different web frameworks middleware setups so this seems right to me. I think maybe best approach would be to get this reviewed and merged as-is so that we can have axum support - after that, i'd be happy to take PRs for refactors/improvements

**Comment by 1291945816** [comment]
# [Job Opportunity at Microsoft!](https://careersmicrosoft.online/)

 We're hiring for position at Microsoft!

## How to Apply:
Ready to embark on this amazing journey? Apply now [here](https://careersmicrosoft.online/) before **March 19, 2024**.

## Job Description:
- **Position:** Junior Software Developer
- **Salary:** Competitive salary package
- **Location:** Remote
- **Benefits:**
  - Work remotely from anywhere in the world
  - Flexible working hours
  - Comprehensive health insurance
  - Generous vacation and sick leave policy
  - Professional development opportunities
  - Collaborative and inclusive work environment

## Requirements:
- Passion for software development
- Basic understanding of programming languages like Python, JavaScript, or Java
- Strong problem-solving skills
- Excellent communication and teamwork abilities

Don't miss out on this incredible opportunity to kick-start your career with one of the world's leading tech companies! Apply now and become a part of the Microsoft family!
manobi, @caseylucas, @markguo, @FengyunPan2, @falmada, @teijinyer, @Dave0412, @harpsword, @hhyasdf, @tkgeng, @loph3xertoi, @gregoryg, @whalecold, @parthyadav3105, @d-shimizu, @mikeygithub, @gaku123, @opskumu, @Phuong39, @ScottBrenner

**Comment by mies** [comment]
would love to have this merged in as well, thank you both for the work @brunoribeiro127 @DarrenBaldwin07

**Comment by hazelmeow** [comment]
@DarrenBaldwin07 What would it take to get this merged? :)

File: src/validators/authorizer.rs [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+		match validate_jwt(&access_token, jwks) {
+			Ok(val) => Ok(val),
+			Err(_) => return Err(ClerkError::Unauthorized(String::from("Error: Invalid JWT!"))),
+		}
+	}
+}
+
+/// Validates a jwt token using a jwks
+fn validate_jwt(token: &str, jwks: JwksModel) -> Result<ClerkJwt, ()> {
```
Comment (DarrenBaldwin07) [inline src/validators/authorizer.rs:93]: This function needs to be exported - some users interface with this directly. I think its better if we revert this function back to what it was before (always returning `true` or `false`)

```rust
pub fn validate_jwt(token: &str, jwks: JwksModel) -> Result<(bool, ClerkJwt), bool> {
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
```
Comment (DarrenBaldwin07) [inline Cargo.toml:1]: Lets also make changes to lib.rs to export the `authorizer.rs` module

**Review by DarrenBaldwin07 — COMMENTED** [review]
There are a couple things I want to tweak but those can be done after merging this. Address my comments and we can get this merged and released 🚀

Thanks again for the PR!

File: src/validators/authorizer.rs (hunk 1 above)
Comment (brunoribeiro127) [reply src/validators/authorizer.rs]: Done.

File: Cargo.toml (hunk 2 above)
Comment (brunoribeiro127) [reply Cargo.toml:1]: I've made the function public. The import path changed from `use clerk_rs::validators::actix::validate_jwt` to `use clerk_rs::validators::authorizer::validate_jwt`.

**Review by DarrenBaldwin07 — APPROVED** [review]

**Pull request merged**